core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -439,12 +439,16 @@ Kokkos::Cuda::initialize WARNING: Cuda is allocating into UVMSpace by default
 //----------------------------------------------------------------------------
 
 using ScratchGrain = Cuda::size_type[Impl::CudaTraits::WarpSize];
-enum { sizeScratchGrain = sizeof(ScratchGrain) };
+static constexpr auto sizeScratchGrain = sizeof(ScratchGrain);
```

Comment:
Define it within `scratch_count` because it is only used there.

Reply:
Note that `m_scratchFlagsCount * sizeScratchGrain` is the expression that was potentially overflowing in the `if` statement. Not sure how this could be fixed here.

Reply:
I guess we should really just error out if we detect that the calculation would overflow since we can't allocate that much memory anyway.